lib/bridge.js

```diff
@@ -17,51 +18,108 @@ var Callback = require('./callback');
-    return new Bridge(ip, port, token);
+    return new Bridge(ip, port, token, options);
   }
 
   this.ip = ip;
   this.port = parseInt(port, 10);
   this.token = token;
+  this.type = 1; // 1 = hardware, 2 = software
+  this.forcePlainToken = options && options.forcePlainToken || null;
+  
   this.delayBetweenRequests = options && options.delayBetweenRequests || 250;
   this.delayer = Promise.resolve();
 
   if (!this.ip || !this.port || !this.token || this.port < 1 || this.port > 65536) {
     throw new Error('Please check the arguments!');
   }
+  
+  // get bridge type
+  this._getBridgeType();
+};
+
+/**
+ * This function retrieves the type of bridge (1 = hardware, 2 = software).
+ *
+ * @param void
+ * @returns void
+ * @private
+ */
+Bridge.prototype._getBridgeType = function _getBridgeType () {
+
+	var self = this;
+	self.delayer = new Promise(resolve => {
+		self._request('info')
+			.then(bridge =>
+			{
+				self.type = bridge.bridgeType;
+			})
+			.catch(err => self.type = 2)
+			.finally(() => resolve(true));
+	});
+};
+
+/**
+ * This function gets the required parameters for the token.
+ *
+ * @param {Boolean}		[forcePlainToken=false]		force to use plain instead of hashed token
+ * @returns {Object}
+ * @private
+ */
+Bridge.prototype._getTokenParams = function _getTokenParams (forcePlainToken) {
+
+  if (forcePlainToken || this.type != 1) {
+    return { token: this.token };
+  }
+  else {
+    var ts = new Date().toISOString().substr(0, 19)+'Z'; // YYY-MM-DDTHH:MM:SSZ
+    var rnr = Math.floor(Math.random() * (65535-0) + 0); // Math.random() * (max - min) + min; // uint16 up to 65535
+    var hash = crypto.createHash('sha256').update(ts + ',' + rnr + ',' + this.token).digest('hex');
+	
+	return {
+		ts: ts,
+		rnr: rnr,
+		hash: hash
+	};
+  }
 };
 
 /**
  * This function requests an action.
  *
  * @param {String}    action                    the name of the action
  * @param {String}    [additionalParameter]     additional get parameters
+ * @param {Boolean}	  [forcePlainToken=false]	force to use plain instead of hashed token
  * @returns {Promise<Object>}
  * @private
  */
-Bridge.prototype._request = function _request (action, additionalParameter) {
-  var url = 'http://' + this.ip + ':' + this.port + '/' + action + '?token=' + this.token;
-  var self = this;
-
-  if (additionalParameter) {
-    url += additionalParameter;
-  }
-
-  return new Promise(function requestWithoutDelay (resolve) {
-    self.delayer = self.delayer
-        .then(function doRequest () {
-          var req = request({
-            uri: url,
-            json: true,
-          });
-
-          resolve(req); // Resolve returned promise without delay
-
-          return req;
-        })
-        .catch(function ignoreErrorsForNextRequest () {})
-        .delay(self.delayBetweenRequests);
-  });
+Bridge.prototype._request = function _request (action, additionalParameter, forcePlainToken) {
+	
+	return new Promise(resolve =>
+	{
```

Comment:
Like already commented, please put the curly brace in the same line as the arrow, everywhere in your changes